### PR TITLE
Add synbols benchmark

### PR DESCRIPTION
**Pull request by prlz77**
1. Adds synbols as an `InMemoryDataset` 
2. Introduces the dependency on `h5py` and `requests` in order to load and download synbols

I added Synbols as a setting in `test_classes_incremental` and it worked. However I don't know if it is the right place to test it, so I have not commited that change.

### CODE REVIEW

File: continuum/datasets/synbols.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        """
+        self.path = path
+        self.task = task
+        self.ratios = ratios
+        print("Loading hdf5...")
+        with h5py.File(path, 'r') as data:
+            self.x = data['x'][...]
+            y = data['y'][...]
+            print("Converting json strings to labels...")
+            with multiprocessing.Pool(8) as pool:
```
Comment (arthurdouillard) [inline continuum/datasets/synbols.py:86]: Could you replace the number `8` by:

```
min(8, multiprocessing.cpu_count())
```

File: continuum/datasets/synbols.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+                self.y = pool.map(json.loads, y)
+            print("Done converting.")
+            if isinstance(mask, str):
+                if "split" in data:
+                    if mask in data['split'] and mask == "random":
+                        self.mask = data["split"][mask][...]
+                    else:
+                        self.mask = self.parse_mask(mask, ratios=ratios)
+                else:
+                    raise ValueError
```
Comment (arthurdouillard) [inline continuum/datasets/synbols.py:96]: Can you provide a understandable error message in the exception. If it is raised, what should the user do?

**Review by arthurdouillard — CHANGES_REQUESTED** [review]
Thank you for your PR!

I've only made a few requests, if you could modify. If you don't have time, I can do it myself if you prefer :)

File: continuum/datasets/synbols.py (hunk 1 above)
Comment (prlz77) [reply continuum/datasets/synbols.py]: Done!

File: continuum/datasets/synbols.py (hunk 2 above)
Comment (prlz77) [reply continuum/datasets/synbols.py]: I simplified the whole thing and the error is no longer necessary

File: continuum/datasets/synbols.py (hunk 1 above)
Comment (arthurdouillard) [reply continuum/datasets/synbols.py]: Maybe use the `min(8, multiprocessing.cpu_count())`? Using all cpu counts may be a bit violent? ahah

File: continuum/datasets/synbols.py (hunk 2 above)
Comment (arthurdouillard) [reply continuum/datasets/synbols.py]: Great thanks

**Pull request merged**